tests/framework/utils_cpu_templates.py

```diff
@@ -24,23 +24,23 @@ def get_supported_cpu_templates():
     Return the list of CPU templates supported by the platform.
     """
     # pylint:disable=too-many-return-statements
-    match cpuid_utils.get_cpu_vendor():
-        case cpuid_utils.CpuVendor.INTEL:
-            # T2CL template is only supported on Cascade Lake and newer CPUs.
-            if global_props.cpu_codename == cpuid_utils.CpuModel.INTEL_SKYLAKE:
-                return sorted(set(INTEL_TEMPLATES) - set(["T2CL"]))
+    vendor = get_cpu_vendor()
+    cpu_model = global_props.cpu_model
+    host_linux = global_props.host_linux_version_tpl
+    cpu_codename = global_props.cpu_codename
+
+    match (vendor, cpu_codename, cpu_model):
+        # T2CL template is only supported on Cascade Lake and newer CPUs.
+        case (CpuVendor.INTEL, CpuModel.INTEL_SKYLAKE, _):
+            return sorted(set(INTEL_TEMPLATES) - set(["T2CL"]))
+        case (CpuVendor.INTEL, _, _):
             return INTEL_TEMPLATES
-        case cpuid_utils.CpuVendor.AMD:
+        case (CpuVendor.AMD, _, _):
             return AMD_TEMPLATES
-        case cpuid_utils.CpuVendor.ARM:
-            if global_props.host_linux_version_tpl < (6, 1):
-                return []
-            match global_props.cpu_model:
-                case cpuid_utils.CpuModel.ARM_NEOVERSE_N1:
-                    return []
-                case cpuid_utils.CpuModel.ARM_NEOVERSE_V1:
-                    return ARM_TEMPLATES
-    return []
+        case (CpuVendor.ARM, _, CpuModel.ARM_NEOVERSE_V1) if host_linux >= (6, 1):
+            return ARM_TEMPLATES
+        case _:
+            return []
```

Comment:
Nit: maybe move this outside the case so the function returns something as a fallback

Reply:
I think they are functionally the same, I'm coming from a rust-like style e.g.
```rust
fn my_function(x: u32) -> Option<bool> {
    match x {
        0 => Some(true),
        3 => Some(false),
        _ => None
    }
}
```

Reply:
Yeah I was just concerned pylint may complain that the function may not return anything, but since it does not seem to complain, I'm fine with it.